src/Ratios.jl

```diff
@@ -34,13 +34,13 @@ convert{T<:Integer, S<:Integer}(::Type{Rational{T}}, r::SimpleRatio{S}) = conver
 -(x::SimpleRatio, y::SimpleRatio) = SimpleRatio(x.num*y.den - x.den*y.num, x.den*y.den)
 ^(x::SimpleRatio, y::Integer) = SimpleRatio(x.num^y, x.den^y)
 
--{T<:Signed}(x::SimpleRatio{T}) = SimpleRatio(-x.num, x.den)
--{T<:Unsigned}(x::SimpleRatio{T}) = throw(OverflowError())
+-(x::SimpleRatio{T}) where {T<:Signed} = SimpleRatio(-x.num, x.den)
+-(x::SimpleRatio{T}) where {T<:Unsigned} = throw(OverflowError("cannot negate unsigned number"))
```

Comment:
Actually this also needs to be addressed in order to keep supporting julia 0.6, with something like

```julia
throw(VERSION < v"0.7.0-DEV.1269" ? OverflowError() : OverflowError("cannot negate unsigned number"))
```